### PR TITLE
SharePoint library add functions and function overloads

### PR DESCRIPTION
**Pull request by glachana**
Adds DownloadFileContent() overloads to SharePoint client
Adds new functions DeleteFile() and DeleteFolder() to SharePoint client

This PR can replace PR #22565 and PR #22402 as discussed with @pri-kise and @jwikman in PR #22402

### CODE REVIEW

File: Modules/System/SharePoint/src/SharePointClientImpl.Codeunit.al
Comment (glachana) [reply Modules/System/SharePoint/src/SharePointClientImpl.Codeunit.al:574]: Well spotted! I forgot about it. I just made a new commit that adds it.

**Comment by jwikman** [comment]
Looking good at first glance, @glachana.

I'm busy for a few more hours, will look through this later today.

**Comment by glachana** [comment]
I just add the equivalent of DownloadFileContent() with ServerRelativeUrl instead of OdataId param to SharePoint client
Also, I add the equivalent of DeleteFile() and DeleteFolder() with ServerRelativeUrl instead of OdataId to SharePoint client

New functions :
-     procedure DownloadFileContentByServerRelativeUrl(ServerRelativeUrl: Text; var FileInStream: InStream): Boolean
-     procedure DownloadFileContentByServerRelativeUrl(ServerRelativeUrl: Text; FileName: Text): Boolean
-     procedure DownloadFileContentByServerRelativeUrl(ServerRelativeUrl: Text; var TempBlob: Codeunit "Temp Blob"): Boolean
-     procedure DeleteFileByServerRelativeUrl(ServerRelativeUrl: Text): Boolean
-     procedure DeleteFolderByServerRelativeUrl(ServerRelativeUrl: Text): Boolean

File: Modules/System/SharePoint/src/SharePointClient.Codeunit.al
Comment (pri-kise) [inline Modules/System/SharePoint/src/SharePointClient.Codeunit.al:343]: Looks like a copy paste error.
Should be `ServerRelativeUrl`

Comment (glachana) [reply Modules/System/SharePoint/src/SharePointClient.Codeunit.al]: Thanks for your feedback. Fixed !! :)

**Review by jwikman — APPROVED** [review]
I think this approach is much better than the X-Http-Method one.

I'll abandon my PRs in advantage for this one.

I don't have any environment to test this out myself right now, so I can only take your words on that it works. :)
I'll be working on another SharePoint feature in one of our apps in a week or two - and I'll be sure to test these out then.

**Comment by glachana** [comment]
Hi All,

Is there a chance to see these changes reviewed and merged into "Master" and published soon?

Have a nice day :)

Thanks

**Comment by jwikman** [comment]
@JesperSchulz ☝️

**Comment by JesperSchulz** [comment]
I was waiting for one more sign-off from a different partner, but let's get on this! Processing PR internally!

**Comment by dsaveyn** [comment]
We are also in desperate need of this. Code seems okay and exactly what we need.

**Review by pri-kise — APPROVED** [review]

**Comment by JesperSchulz** [comment]
22.2 will be the likely release for this 🥳

**Review by JesperSchulz — APPROVED** [review]

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 10731.

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 22.

Build ID to track: 56517.

**Comment by glachana** [comment]
@JesperSchulz @pri-kise,

Hi all, 
I just download symbols with build id 56517, and it seems that DeleteFile and DeleteFileByServerRelativeUrl public implementation in codeunit 9100 "SharePoint Client" was not merged into the main.

Is this normal ?

**Comment by JesperSchulz** [comment]
I think we ran into some sort of merge issues. I'll make sure we get that fixed for 22.3. I'll also investigate what when wrong...

**Pull request merged**
